### ISSUE TITLE
When training YOLOv8, I am unable to properly pass the parameters for model.train().

### ISSUE THREAD

**Issue by chenxin03** [ultralytics/ultralytics#10101]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

When I using model.train(), it looks like only "data" and "epochs" are correctly passed, while other parameters revert to default.

In fact, when I run this 2hours ago, it's default save path is 'runs/classify', but I don't know why it changed to 'Classify-aug-test', and even can't get almost all the train params (like amp, 1650ti can't run with amp=True) 

I tried else model like yolov8m-cls, yolov8n, the result is the same.

Could you tell me what I did wrong? Thank you very much😉.

### Environment

Ultralytics YOLOv8.1.47 🚀 Python-3.11.4 torch-2.2.1+cu121 CUDA:0 (NVIDIA GeForce GTX 1650 Ti, 4096MiB)
Setup complete ✅ (8 CPUs, 15.8 GB RAM, 479.0/931.5 GB disk)

### Minimal Reproducible Example

there is my train code:
```
from ultralytics import YOLO
if __name__ == '__main__':
    model = YOLO('yolov8n-cls.pt')

    results = model.train(
        data=r'data/',
        epochs=200,
        imgsz=640,
        name='test2',
        # exist_ok=True,
        batch=-1,
        workers=0,
        device='cuda:0',
        amp=False,
        cos_lr=True,
        # mosaic=0,
        # close_mosaic=0,
        fliplr=0,
        label_smoothing=0.01,
        save=True,
        save_period=25,
        patience=25,
        resume=True,
        plots=True
    )
```
when I run it, yolo print:

> Ultralytics YOLOv8.1.47 🚀 Python-3.11.4 torch-2.2.1+cu121 CUDA:0 (NVIDIA GeForce GTX 1650 Ti, 4096MiB)
engine\trainer: task=classify, mode=train, model=yolov8n-cls.pt, data=data/, epochs=200, time=None, patience=50, batch=-1, imgsz=640, save=True, save_period=-1, cache=False, device=cuda:0, workers=16, project=Classify-aug-test, name=yolov8n-cls3, exist_ok=False, pretrained=True, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=True, close_mosaic=10, resume=yolov8n-cls.pt, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.2, lrf=0.01, momentum=0.9, weight_decay=0.0001, warmup_epochs=0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.1, nbs=256, hsv_h=0.015, hsv_s=0.4, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, boxes=True, random_erasing=0.4, crop_percentage=1.0, save_dir=Classify-aug-test\yolov8n-cls3

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@chenxin03 hey there! 😊 It seems like there's a bit of confusion with how parameters are being recognized during the `model.train()` call. Your setup and the provided code snippet look fine, so this behavior is indeed unusual.

One thing to note is that when you specify parameters in the `model.train()` function, they should be directly recognized and used by the training process. However, if some parameters are reverting to their defaults or not being acknowledged, this could indicate a potential issue with how the parameters are parsed or a bug within the version of Ultralytics YOLO you're using.

Considering your situation, here are a couple of things you might want to try:
1. **Ensure Compatibility**: Double-check that the version of Ultralytics YOLO you're using is up to date or compatible with your setup. Occasionally, bugs are fixed in newer versions that might resolve your issue.
2. **Use Absolute Path**: While not directly related to your issue, using absolute paths can sometimes help avoid path-related issues. For example: `data='/absolute/path/to/data/'`.

If these suggestions do not resolve your issue, it might be helpful to raise this behavior with the Ultralytics team as you've already searched for similar issues. Sharing your detailed observation, as you did here, helps in diagnosing the problem more effectively.

Since you're open to contributing by submitting a PR, it would be great to look into the parameter parsing section of the training module if you're comfortable diving into the code. Maybe there's a small tweak needed there to ensure all parameters are correctly used.

Thanks for bringing this to attention and for your willingness to contribute to improving YOLOv8! Keep exploring and coding. 👍

**Comment by Burhan-Q** [comment]
Additionally, the NVIDIA GeForce GTX 1650 Ti GPUs are notoriously problematic with PyTorch. You can find several reports of issues with these GPUs and I would not recommend them for training.
